Hunting Queries/SecurityEvent/RareProcessPath.yaml

```diff
@@ -0,0 +1,66 @@
+id: 0ff22697-dc58-4623-b844-a767629840cd
+name: Rare Process Path
+description: |
+  'Identifies when a process is running from a rare path. This could indicate malicious or unexpected activity as attacks 
+  often try to use common process names running from non-standard locations'
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+tactics:
+  - Execution
+query: |
+
+  let end = startofday(now());
+  let start = end - 7d;
+  let processEvents=
+  SecurityEvent
+  | where TimeGenerated >= start and TimeGenerated <= end
+  | where EventID==4688
+  // excluding well known processes
+  | where NewProcessName !endswith ':\\Windows\\System32\\conhost.exe' and ParentProcessName !endswith ':\\Windows\\System32\\conhost.exe'
+  | where ParentProcessName !endswith ":\\Windows\\System32\\wuauclt.exe" and NewProcessName !startswith "C:\\Windows\\SoftwareDistribution\\Download\\Install\\AM_Delta_Patch_"
+  | where NewProcessName !has ":\\Windows\\WinSxS\\amd64_microsoft-windows-servicingstack_" and ParentProcessName !has ":\\Windows\\WinSxS\\amd64_microsoft-windows-servicingstack_"
+  | where NewProcessName !endswith ":\\WindowsAzure\\SecAgent\\WaSecAgentProv.exe" 
+  //| where ParentProcessName !has ":\\WindowsAzure\\GuestAgent_" and NewProcessName !has ":\\WindowsAzure\\GuestAgent_"
+  | where ParentProcessName !has ":\\WindowsAzure\\WindowsAzureNetAgent_" and NewProcessName !has ":\\WindowsAzure\\WindowsAzureNetAgent_"
+  | where ParentProcessName !has ":\\ProgramData\\Microsoft\\Windows Defender\\platform\\" and ParentProcessName !endswith "\\MpCmdRun.exe" 
+  | where NewProcessName !has ":\\ProgramData\\Microsoft\\Windows Defender\\platform\\" and NewProcessName !endswith "\\MpCmdRun.exe" 
+  | where NewProcessName !has ':\\Program Files\\Microsoft Monitoring Agent\\Agent\\'
+  // filter out common randomly named paths and files
+  | where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{3}\.tmp")
+  | where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{4}\.tmp")
```

Comment:
Could compress this to:
| where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{3,4}\.tmp"

Reply:
True, simply pulled this from another query that Tim did awhile back.  I am not sure I will change this, while I can't see the negative impact, I don't want to have to test again across multiple data sets to verify as I don't think there is much perf hit.